redis/src/main/scala/zio/redis/Input.scala

```diff
@@ -409,11 +409,19 @@ object Input {
     def encode(data: Update): Chunk[RespValue.BulkString] = Chunk.single(encodeString(data.stringify))
   }
 
-  final case class Varargs[-A](input: Input[A]) extends Input[Iterable[A]] {
+  final case class Varargs[-A](input: Input[A]) extends Input[Iterable[A]] { // todo: never used ??
     def encode(data: Iterable[A]): Chunk[RespValue.BulkString] =
       data.foldLeft(Chunk.empty: Chunk[RespValue.BulkString])((acc, a) => acc ++ input.encode(a))
   }
 
+  case class EvalInput[K, A](keysInput: Input[K], argsInput: Input[A]) extends Input[Script[K, A]] {
```

Comment:
I don't think this is correct. What you have in `Script` should be specified as an input, i.e.:
- string script
- varargs keys
- varargs args

Also, be consistent and make it final.

Reply:
Hopefully, I understood you right and did how you wanted)